lang/ghc/Portfile

```diff
@@ -29,144 +24,145 @@ long_description    \
+        xinstall -m 0755 -d ${workpath}/src
+        system -W ${workpath}/src \
+            "tar xJf ${distpath}/${name}-${version}-src${extract.suffix}"
+        # fix DejaVu texlive fonts names, e.g. "\setmonofont{DejaVu Sans Mono}"
+        foreach f "\
+            ${workpath}/src/${name}-${version}/libraries/Cabal/Cabal/doc/conf.py \
+            ${workpath}/src/${name}-${version}/docs/users_guide/conf.py \
+            " {
+            reinplace -E "s|(\{DejaVu\[\[:alnum:\]_\]*)\[\[:space:\]\]+|\\1|g" \
+                ${f}
+            reinplace -E "s|(\{DejaVu\[\[:alnum:\]_\]*)\[\[:space:\]\]+|\\1|g" \
+                ${f}
+            reinplace -E "s|\{(DejaVu\[\[:alnum:\]_\]*)\}|\{\\1.ttf\}|g" \
+                ${f}
+        }
+        # fix python3 executable name
+        foreach f "\
+            ${workpath}/src/${name}-${version}/boot \
+            ${workpath}/src/${name}-${version}/utils/checkUniques/check-uniques.py \
+            " {
+            reinplace -E "s|^(#!/usr/bin/env python)3|\\1${python3_version}|" \
+                ${f}
         }
+        system -W ${workpath}/src/${name}-${version} \
+            "export PATH=/usr/bin:\${PATH} CC=/usr/bin/gcc ; \
+                ./boot"
+        system -W ${workpath}/src/${name}-${version} \
+            "export PATH=/usr/bin:\${PATH}:${destroot}${prefix}/bin CC=/usr/bin/gcc ; \
+                ./configure --prefix=${destroot}${prefix}"
+        system -W ${workpath}/src/${name}-${version} \
+            "export PATH=/usr/bin:\${PATH} CC=/usr/bin/gcc ; \
+                make -j${buildmakejobs}"
+        system -W ${workpath}/src/${name}-${version} \
+            "export PATH=/usr/bin:\${PATH} CC=/usr/bin/gcc ; \
+                make install -j${buildmakejobs}"
     }
 
-    # Walk the current package.conf.d and delete all files not belonging to any
-    # port, since haskell packages now directly drop it in packages.conf.d and
-    # register the file to themselves. The package cache is updated by the
-    # post-activate phase of the packages, which now runs ghc-pkg recache.
-    #
-    # This *will* cause pre-deactivation failures for all ports that still are
-    # correctly installed and would unregister themselves on uninstall.
-    # Unfortunately there is no good way around this now that the contents of
-    # PortGroups are stored together with Portfiles and used for deactivation
-    # (otherwise I could just edit the PortGroup and make unregistration
-    # failures non-fatal)
-    foreach configfile [glob -nocomplain -directory $libprefix/package.conf.d *] {
-        if {[registry_file_registered $configfile] == 0} {
-            # file is not registered to any port, remove it
-            catch {delete $configfile}
+    # delete any destroot path appearing in text files
+    fs-traverse f ${destroot}${prefix} {
+        if {[file isfile ${f}]} {
+            set textfile_flag [ exec /bin/sh -c "\[\[ \
```

Comment:
I don't think starting a shell for every file is a good idea, especially since this code is quite hard to read. Why not instead a Tcl-based implementation, e.g.:

```tcl
fs-traverse f ${destroot}${prefix} {
    if {[file isfile ${f}]} {
        if {[string match "text/*" [lindex [exec /usr/bin/file --mime-type ${f}] end]]} {
            reinplace "s|${destroot}||g" ${f}
        }
    }
}
```

Apart from that, I'm really wondering whether GHC doesn't support the DESTDIR environment variable in its Makefile so that this entire reinplace dance could possibly be avoided. Remember that binaries may still contain references to the ${destroot}, which we also wouldn't want.

Reply:
Great I'll swap in your tcl code—thanks.

Reply:
I've checked the binaries using both `otool` and `emacs`. There's no `${destroot}`.

The entire Haskell development toolbox is missing `DESTDIR`. See:
*https://github.com/commercialhaskell/stack/issues/848
* https://github.com/ndmitchell/hlint/issues/699
* https://github.com/gibiansky/IHaskell/issues/1061
* https://github.com/commercialhaskell/stack/issues/848#issuecomment-508965440